File: 백준/Bronze/2747.피보나치 수/피보나치 수.cc
#include <iostream>
#include <vector>

using namespace std;

int Fibonacci(int n);

int main()
{
    int num;
    cin >> num;
    
    cout << Fibonacci(num) ;
}

int Fibonacci(int n)
{
    vector<int> Fibonacci;
    
    Fibonacci.push_back(0);
    Fibonacci.push_back(1);
    
    for(int i = 2; i <= n; i++)
        Fibonacci.push_back(Fibonacci[i-2] + Fibonacci[i-1]);
    
    return Fibonacci[n];
}

File: 프로그래머스/2/12945. 피보나치 수/피보나치 수.java
class Solution {
    public int solution(int n) {
        
        long[] fibonacci = new long [100001];
        fibonacci[0] = 0; fibonacci[1] = 1;
        
        for(int i = 2; i <= 100000; i++){
            fibonacci[i] = (fibonacci[i - 1] + fibonacci[i - 2]) % 1234567;
            
            if(i == n) {
                return (int)fibonacci[n];
            }
        }
        
        
        return 0;
    }
}